### PR TITLE
Added driver auto generation directions

### PR DESCRIPTION
**Pull request by sistillson**

### CODE REVIEW

**Comment by richardxia** [comment]
I pushed a couple commits to get this to line up more with the current version of the library and the interfaces. If you guys could take a look and provide feedback, that'd be great.

Also, I completely changed the example that used to show how to set a CFLAG to parameterize the base address of the PIO block because that doesn't make sense anymore with the driver code and autogenerated BSP. https://github.com/sifive/block-pio-sifive/pull/22/commits/79a015fee3f13d42c6ea9c28524235377fed734e Please take a look at that as well.

File: README.md [hunk 1]
```diff
@@ -573,30 +573,124 @@ def pioHook =
...
+
+publish driverImplementations = pioDriver, Nil
+```
+
+
 ## Making a test
 
 Currently, only C integration tests are supported. A simple test is included
-with this repository in `block-pio-sifive/tests/demo/main.c`. It looks like this.
+with this repository in `block-pio-sifive/tests/demo/main.c`. It looks similar to this.
```
Comment (albertchen-sifive) [inline README.md]: it's no longer included in the `preonboarded` branch so I would just tell them to copy the code block to `block-pio-sifive/tests/demo/main.c`

File: README.md [hunk 2]
```diff
@@ -642,39 +736,35 @@ global def demo =
   makeTestProgramPlan programName cFiles
 ```
 
-Suppose we did not want to hardcode the address of the pio block and we edited
+Suppose we did not want to hardcode the pin number of being tested and we edited
```
Comment (albertchen-sifive) [inline README.md]: get rid of the "of"

**Review by albertchen-sifive — COMMENTED** [review]
Just realized we should probably also include a `wit-manifest.json` in this branch and instructions on how to actually checkout all these tools and stuff.

**Review by albertchen-sifive — APPROVED** [review]

**Pull request merged**